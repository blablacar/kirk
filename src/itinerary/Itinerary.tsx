import React, { Fragment } from 'react'
import cc from 'classcat'
import isEmpty from 'lodash.isempty'
import isString from 'lodash.isstring'

import { ItineraryCollapsible } from '../_internals/itineraryCollapsible'
import { computeKeyFromPlace, ItineraryLocation } from '../_internals/itineraryLocation'
import { color } from '../_utils/branding'
import { A11yProps, pickA11yProps } from '../_utils/interfaces'
import { Place } from '../_utils/place'
import { Bullet, BulletTypes } from '../bullet'
import { SpacingDivider } from '../divider/spacingDivider'
import { SubHeader } from '../subHeader'
import { Text, TextDisplayType, TextTagType } from '../text'
import { StyledItinerary } from './Itinerary.style'

export type ItineraryProps = A11yProps &
  Readonly<{
    places?: Place[]
    className?: string
    fromAddon?: string
    toAddon?: string
    fromAddonAriaLabel?: string
    toAddonAriaLabel?: string
    small?: boolean
    headline?: string
    highlightRoad?: boolean
    isCollapsible?: boolean
    collapsedLabel?: string
    segmentCollapsedLabels?: Array<string>
    collapsedAriaProps?: A11yProps
    segments?: Array<Array<Place>>
  }>

type StopoverProps = Readonly<{
  isCollapsible?: boolean
  intermediatePlaces: Array<Place>
  collapsedLabel: string
  collapsedAriaProps?: A11yProps
  small: boolean
  withTime: boolean
}>

interface RootA11yProps {
  'aria-label'?: string
  'aria-labelledby'?: string
}

const isNonEmptyString = (str: string) => isString(str) && str.trim().length > 0

const computeRootA11yProps = (ariaLabel?: string, ariaLabelledBy?: string): RootA11yProps => {
  const rootA11yProps: RootA11yProps = {}
  if (ariaLabel) {
    rootA11yProps['aria-label'] = ariaLabel
  }
  const ariaLabelledByValue = isNonEmptyString(ariaLabel) ? null : ariaLabelledBy
  if (ariaLabelledByValue) {
    rootA11yProps['aria-labelledby'] = ariaLabelledByValue
  }
  return rootA11yProps
}

// Get places between departure and arrival
const getIntermediatePlaces = (places: Place[]) => places.slice(1, -1)

const STOPOVER_COUNT_TO_COLLAPSE_FROM = 1

const renderLocation = (
  places: Place[],
  isArrival: boolean,
  small: boolean,
  withTime: boolean,
  hasBottomAddon: boolean,
  displayConnection?: boolean,
) => {
  // if there's only one place, we display it as Arrival, not Departure
  if (places.length === 1 && !isArrival) {
    return null
  }

  const place = isArrival ? places[places.length - 1] : places[0]

  return (
    <ItineraryLocation
      place={place}
      isArrival={isArrival}
      hasTime={!small && withTime}
      hasSubLabel={!small && !isEmpty(place.subLabel)}
      hasBottomAddon={isArrival ? hasBottomAddon : false}
      displayConnection={displayConnection}
    />
  )
}

const renderAddon = (type: string, addon: string, ariaLabel: string) => {
  if (!isNonEmptyString(addon) || (type !== 'from' && type !== 'to')) {
    return null
  }

  return (
    <li
      className={`kirk-itineraryLocation kirk-itinerary-addon kirk-itinerary-addon--${type}`}
      aria-label={ariaLabel}
    >
      <div className="kirk-itineraryLocation-roadContainer" aria-hidden="true">
        <Bullet className="kirk-itineraryLocation-bullet" type={BulletTypes.ADDON} />
        {type === 'from' && <div className="kirk-itineraryLocation-road" aria-hidden="true" />}
      </div>
      <div className="kirk-itineraryLocation-label">
        <Text tag={TextTagType.PARAGRAPH} display={TextDisplayType.CAPTION} textColor={color.gray}>
          {addon}
        </Text>
      </div>
    </li>
  )
}

const renderStopover = ({
  isCollapsible,
  intermediatePlaces,
  collapsedLabel,
  collapsedAriaProps,
  small,
  withTime,
}: StopoverProps) => {
  if (isCollapsible) {
    if (intermediatePlaces.length > STOPOVER_COUNT_TO_COLLAPSE_FROM) {
      return (
        <ItineraryCollapsible
          places={intermediatePlaces}
          label={collapsedLabel}
          {...collapsedAriaProps}
        />
      )
    }
    return intermediatePlaces.map(place => (
      <ItineraryLocation
        place={place}
        displaySubLabelOnly
        isSmall
        className="kirk-itineraryLocation-smallLabel"
        key={`${computeKeyFromPlace(place)}`}
      />
    ))
  }
  return intermediatePlaces.map(place => (
    <ItineraryLocation
      place={place}
      hasTime={!small && withTime}
      hasSubLabel={!small && Boolean(place.subLabel)}
      key={computeKeyFromPlace(place)}
    />
  ))
}

export const Itinerary = (props: ItineraryProps) => {
  const {
    className,
    places = [],
    fromAddon,
    toAddon,
    fromAddonAriaLabel,
    toAddonAriaLabel,
    small = false,
    headline = null,
    highlightRoad = true,
    isCollapsible = false,
    collapsedLabel,
    collapsedAriaProps,
    segments,
    segmentCollapsedLabels,
  } = props
  const a11yAttrs = pickA11yProps<ItineraryProps>(props)

  let withTime = true
  // Add the small class if we don't have "time" to prevent empty content
  // This is a quick fix and only applies for places prop not for segments.
  if (!isEmpty(places)) {
    withTime = places.filter(p => !isEmpty(p.time)).length > 0
  }

  // Remove aria-labelledby attribute if aria-label already used
  const rootA11yProps = computeRootA11yProps(a11yAttrs['aria-label'], a11yAttrs['aria-labelledby'])

  const intermediatePlaces = getIntermediatePlaces(places)

  return (
    <StyledItinerary className={cc(['kirk-itinerary-root', className])} {...rootA11yProps}>
      {isNonEmptyString(headline) && (
        <Fragment>
          <SubHeader>{headline}</SubHeader>
          <SpacingDivider />
        </Fragment>
      )}
      <ul
        className={cc([
          {
            'kirk-itinerary--noTime': small || !withTime,
            'kirk-itinerary--highlightRoad': highlightRoad,
          },
        ])}
      >
        {renderAddon('from', fromAddon, fromAddonAriaLabel)}
        {isEmpty(segments) ? (
          <Fragment>
            {renderLocation(places, false, small, withTime, false)}
            {renderStopover({
              isCollapsible,
              intermediatePlaces,
              collapsedLabel,
              collapsedAriaProps,
              small,
              withTime,
            })}
            {renderLocation(places, true, small, withTime, isNonEmptyString(toAddon))}
          </Fragment>
        ) : (
          segments.map((segmentPlaces, index) => {
            const segmentIntermediatePlaces = getIntermediatePlaces(segmentPlaces)
            const displayConnection = segments.length !== index + 1
            return (
              <Fragment>
                {renderLocation(segmentPlaces, false, small, withTime, false)}
                {renderStopover({
                  isCollapsible,
                  intermediatePlaces: segmentIntermediatePlaces,
                  collapsedLabel: segmentCollapsedLabels[index],
                  collapsedAriaProps,
                  small,
                  withTime,
                })}
                {renderLocation(
                  segmentPlaces,
                  true,
                  small,
                  withTime,
                  isNonEmptyString(toAddon),
                  displayConnection,
                )}
              </Fragment>
            )
          })
        )}

        {renderAddon('to', toAddon, toAddonAriaLabel)}
      </ul>
    </StyledItinerary>
  )
}
