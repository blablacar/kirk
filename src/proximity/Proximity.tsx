import React from 'react'
import cc from 'classcat'

import { color } from '_utils/branding'
import ProximityIcon from 'icon/proximityIcon'

export enum Distances {
  NONE = 'NONE',
  CLOSE = 'CLOSE',
  MIDDLE = 'MIDDLE',
  FAR = 'FAR',
}

const size = '20px'

const getColorAndTitle = (index: string, value: string, title: string) => {
  const defaultParams = { title: '', iconColor: color.lightGray }
  switch (value) {
    case Distances.CLOSE:
      return index === value
        ? {
            title: title || defaultParams.title,
            iconColor: color.green,
          }
        : defaultParams

    case Distances.MIDDLE:
      return index === value
        ? {
            title: title || defaultParams.title,
            iconColor: color.yellow,
          }
        : defaultParams

    case Distances.FAR:
      return index === value
        ? {
            title: title || defaultParams.title,
            iconColor: color.orange,
          }
        : defaultParams

    default:
      return defaultParams
  }
}

export interface ProximityProps {
  readonly className?: string
  readonly value: Distances
  readonly title?: string
}

const Proximity = ({ value, title, className }: ProximityProps) => (
  <div className={cc(['kirk-proximity', className])}>
    {[Distances.CLOSE, Distances.MIDDLE, Distances.FAR].map(distance => (
      <ProximityIcon key={distance} size={size} {...getColorAndTitle(distance, value, title)} />
    ))}
  </div>
)

export default Proximity
