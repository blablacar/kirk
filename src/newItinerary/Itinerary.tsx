import React from 'react'

import { TheVoice } from '../theVoice'
import { StyledItinerary } from './Itinerary.style'
import { NormalizeProps } from '../layout/layoutNormalizer'

export enum Lines {
  NONE = 'none',
  INACTIVE = '#DDD',
  ACTIVE = '#054752',
  CONNECTION = 'connection',
  HIDDEN_STOPS = 'hidden_stops_top',
  HIDDEN_STOPS_BOTTOM = 'hidden_stops_bottom',
}

export type ItineraryProps = NormalizeProps & Readonly<{
  children: React.ReactNode
  headline?: string
  small?: boolean
}>

export const Itinerary = ({ children, headline, small = false }: ItineraryProps) => (
  <StyledItinerary small={small}>
    {headline && <TheVoice>{headline}</TheVoice>}
    <ul>{children}</ul>
  </StyledItinerary>
)
