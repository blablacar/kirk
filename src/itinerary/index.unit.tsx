import React from 'react'
import renderer from 'react-test-renderer'

import Itinerary from 'itinerary'
import Proximity from 'proximity'

const places = [
  {
    distanceFromPoint: '1,5km',
    time: '09:00',
    isoDate: '2017-12-11T09:00',
    subLabel: <Proximity value="FAR" title="Pick up point is quite far fom your place"/>,
    mainLabel: 'Paris',
  },
  {
    time: '12:00',
    isoDate: '2017-12-11T12:00',
    subLabel: <Proximity value="FAR" title="Pick up point is quite far fom your place"/>,
    mainLabel: 'Tours',
  },
  {
    distanceFromPoint: '8km',
    time: '15:00',
    isoDate: '2017-12-11T15:00',
    subLabel: <Proximity value="FAR" title="Pick up point is quite far fom your place"/>,
    mainLabel: 'Bordeaux',
  },
]

describe('Itinerary component', () => {
  it('Render with custom className', () => {
    const itinerary = shallow((
      <Itinerary className="test" places={places} />
    ))
    expect(itinerary.hasClass('test')).toBe(true)
  })

  it('Should display the top addon', () => {
    const itinerary = shallow((
      <Itinerary fromAddon="test" places={places} />
    ))
    expect(itinerary.find('.kirk-itinerary-fromAddon').exists()).toBe(true)
  })

  it('Should display the bottom addon', () => {
    const itinerary = shallow((
      <Itinerary toAddon="test" places={places} />
    ))
    expect(
      itinerary.find('.kirk-itinerary--arrival').hasClass('kirk-itinerary-location--toAddon'),
    ).toBe(true)
    expect(itinerary.find('.kirk-itinerary-toAddon').exists()).toBe(true)
  })

  it('Should display stopover', () => {
    const itinerary = shallow((
      <Itinerary places={places} />
    ))
    expect(itinerary.find('.kirk-itinerary-location')).toHaveLength(places.length)
  })

  it('Should display link', () => {
    const itinerary = shallow((
      <Itinerary places={[
        {
          distanceFromPoint: '1,5km',
          time: '09:00',
          isoDate: '2017-12-11T09:00',
          subLabel: <Proximity value="FAR" title="Pick up point is quite far fom your place"/>,
          mainLabel: 'Paris',
          href: '#test',
        },
        {
          distanceFromPoint: '8km',
          time: '15:00',
          isoDate: '2017-12-11T15:00',
          subLabel: <Proximity value="FAR" title="Pick up point is quite far fom your place"/>,
          mainLabel: 'Bordeaux',
        },
      ]} />
    ))
    expect(itinerary.find('.kirk-itinerary-location-wrapper').first().type()).toEqual('a')
    expect(itinerary.find('.kirk-itinerary-location-chevron').exists()).toBe(true)
  })

  it('Should display button', () => {
    const itinerary = shallow((
      <Itinerary places={[
        {
          distanceFromPoint: '1,5km',
          time: '09:00',
          isoDate: '2017-12-11T09:00',
          subLabel: <Proximity value="FAR" title="Pick up point is quite far fom your place"/>,
          mainLabel: 'Paris',
          href: <button type="button" />,
        },
        {
          distanceFromPoint: '8km',
          time: '15:00',
          isoDate: '2017-12-11T15:00',
          subLabel: <Proximity value="FAR" title="Pick up point is quite far fom your place"/>,
          mainLabel: 'Bordeaux',
        },
      ]} />
    ))
    expect(itinerary.find('.kirk-itinerary-location-wrapper').first().type()).toEqual('button')
    expect(itinerary
      .find('.kirk-itinerary-location-wrapper')
      .first()
      .prop('type'))
    .toEqual('button')
    expect(itinerary.find('.kirk-itinerary-location-chevron').exists()).toBe(true)
  })

  it('Should display proximity from departure point', () => {
    const itinerary = renderer.create(
      <Itinerary places={places} />,
    ).toJSON()
    expect(itinerary).toMatchSnapshot()
  })
})
