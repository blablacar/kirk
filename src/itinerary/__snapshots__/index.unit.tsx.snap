// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Itinerary component Should display proximity from departure point 1`] = `
.c0 {
  margin: 0;
  font-weight: 400;
}

.c0.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c0.kirk-text-body,
.c0.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c0.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c0.kirk-text-title,
.c0.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c0.kirk-text-display1 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
}

.c0.kirk-text-display2 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
}

.c0.kirk-text-subheader,
.c0.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c0.kirk-text-title,
.c0.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c0.kirk-text-bodyStrong,
.c0.kirk-text-subheaderStrong,
.c0.kirk-text-titleStrong {
  font-weight: 500;
}

.c2 {
  display: inline-block;
  position: relative;
}

.c2 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c1 {
  display: inline-block;
  line-height: 0;
}

.c1 svg {
  margin-top: 4px;
}

.c1 svg + svg {
  margin-left: 4px;
}

<ul
  className="jsx-1448366035 "
>
  <li
    className="jsx-1448366035 kirk-itinerary-location kirk-itinerary--departure"
    itemProp="location"
    itemScope={true}
    itemType="http://schema.org/Place"
  >
    <meta
      className="jsx-1448366035"
      content="Paris"
      itemProp="name"
    />
    <meta
      className="jsx-1448366035"
      content="Paris"
      itemProp="address"
    />
    <div
      className="jsx-1448366035 kirk-itinerary-location-wrapper"
    >
      <time
        className="jsx-1448366035"
        dateTime="2017-12-11T09:00"
      >
        <span
          className="kirk-text kirk-text-titleStrong c0"
        >
          09:00
        </span>
      </time>
      <div
        className="jsx-1448366035 kirk-itinerary-location-city"
      >
        <span
          className="kirk-text kirk-text-titleStrong c0"
        >
          Paris
        </span>
        <div
          className="jsx-1448366035"
        >
          <div
            className="kirk-proximity c1"
          >
            <svg
              aria-hidden={true}
              className="kirk-icon c2"
              fill="#EDEDED"
              height="20px"
              viewBox="0 0 20 20"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <path
                d="M10 20C4.477 20 0 15.523 0 10S4.477 0 10 0s10 4.477 10 10-4.477 10-10 10zm1.202-14.863a1.568 1.568 0 1 0 0-3.136 1.568 1.568 0 0 0 0 3.136zm.94 1.868h-.005a1.568 1.568 0 0 0-1.563-1.554h-.007c-2.252 0-3.896 1.32-4.288 2.692L5.5 11.051H6.926l.723-2.72 1.166-.825-.874 3.264L6.003 18h1.569l1.83-5.661 1.526 1.269.314 4.392h1.528v-5.02l-1.757-1.756.473-1.767a3.753 3.753 0 0 0 2.853 1.313h.314V9.515a2.51 2.51 0 0 1-2.51-2.51z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="kirk-icon c2"
              fill="#EDEDED"
              height="20px"
              viewBox="0 0 20 20"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <path
                d="M10 20C4.477 20 0 15.523 0 10S4.477 0 10 0s10 4.477 10 10-4.477 10-10 10zm1.202-14.863a1.568 1.568 0 1 0 0-3.136 1.568 1.568 0 0 0 0 3.136zm.94 1.868h-.005a1.568 1.568 0 0 0-1.563-1.554h-.007c-2.252 0-3.896 1.32-4.288 2.692L5.5 11.051H6.926l.723-2.72 1.166-.825-.874 3.264L6.003 18h1.569l1.83-5.661 1.526 1.269.314 4.392h1.528v-5.02l-1.757-1.756.473-1.767a3.753 3.753 0 0 0 2.853 1.313h.314V9.515a2.51 2.51 0 0 1-2.51-2.51z"
              />
            </svg>
            <svg
              aria-hidden={false}
              className="kirk-icon c2"
              fill="#F78B00"
              height="20px"
              viewBox="0 0 20 20"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Pick up point is quite far fom your place
              </title>
              <path
                d="M10 20C4.477 20 0 15.523 0 10S4.477 0 10 0s10 4.477 10 10-4.477 10-10 10zm1.202-14.863a1.568 1.568 0 1 0 0-3.136 1.568 1.568 0 0 0 0 3.136zm.94 1.868h-.005a1.568 1.568 0 0 0-1.563-1.554h-.007c-2.252 0-3.896 1.32-4.288 2.692L5.5 11.051H6.926l.723-2.72 1.166-.825-.874 3.264L6.003 18h1.569l1.83-5.661 1.526 1.269.314 4.392h1.528v-5.02l-1.757-1.756.473-1.767a3.753 3.753 0 0 0 2.853 1.313h.314V9.515a2.51 2.51 0 0 1-2.51-2.51z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </li>
  <li
    className="jsx-1448366035 kirk-itinerary-location"
    itemProp="location"
    itemScope={true}
    itemType="http://schema.org/Place"
  >
    <meta
      className="jsx-1448366035"
      content="Tours"
      itemProp="name"
    />
    <meta
      className="jsx-1448366035"
      content="Tours"
      itemProp="address"
    />
    <div
      className="jsx-1448366035 kirk-itinerary-location-wrapper"
    >
      <time
        className="jsx-1448366035"
        dateTime="2017-12-11T12:00"
      >
        <span
          className="kirk-text kirk-text-titleStrong c0"
        >
          12:00
        </span>
      </time>
      <div
        className="jsx-1448366035 kirk-itinerary-location-city"
      >
        <span
          className="kirk-text kirk-text-titleStrong c0"
        >
          Tours
        </span>
        <div
          className="jsx-1448366035"
        >
          <div
            className="kirk-proximity c1"
          >
            <svg
              aria-hidden={true}
              className="kirk-icon c2"
              fill="#EDEDED"
              height="20px"
              viewBox="0 0 20 20"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <path
                d="M10 20C4.477 20 0 15.523 0 10S4.477 0 10 0s10 4.477 10 10-4.477 10-10 10zm1.202-14.863a1.568 1.568 0 1 0 0-3.136 1.568 1.568 0 0 0 0 3.136zm.94 1.868h-.005a1.568 1.568 0 0 0-1.563-1.554h-.007c-2.252 0-3.896 1.32-4.288 2.692L5.5 11.051H6.926l.723-2.72 1.166-.825-.874 3.264L6.003 18h1.569l1.83-5.661 1.526 1.269.314 4.392h1.528v-5.02l-1.757-1.756.473-1.767a3.753 3.753 0 0 0 2.853 1.313h.314V9.515a2.51 2.51 0 0 1-2.51-2.51z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="kirk-icon c2"
              fill="#EDEDED"
              height="20px"
              viewBox="0 0 20 20"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <path
                d="M10 20C4.477 20 0 15.523 0 10S4.477 0 10 0s10 4.477 10 10-4.477 10-10 10zm1.202-14.863a1.568 1.568 0 1 0 0-3.136 1.568 1.568 0 0 0 0 3.136zm.94 1.868h-.005a1.568 1.568 0 0 0-1.563-1.554h-.007c-2.252 0-3.896 1.32-4.288 2.692L5.5 11.051H6.926l.723-2.72 1.166-.825-.874 3.264L6.003 18h1.569l1.83-5.661 1.526 1.269.314 4.392h1.528v-5.02l-1.757-1.756.473-1.767a3.753 3.753 0 0 0 2.853 1.313h.314V9.515a2.51 2.51 0 0 1-2.51-2.51z"
              />
            </svg>
            <svg
              aria-hidden={false}
              className="kirk-icon c2"
              fill="#F78B00"
              height="20px"
              viewBox="0 0 20 20"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Pick up point is quite far fom your place
              </title>
              <path
                d="M10 20C4.477 20 0 15.523 0 10S4.477 0 10 0s10 4.477 10 10-4.477 10-10 10zm1.202-14.863a1.568 1.568 0 1 0 0-3.136 1.568 1.568 0 0 0 0 3.136zm.94 1.868h-.005a1.568 1.568 0 0 0-1.563-1.554h-.007c-2.252 0-3.896 1.32-4.288 2.692L5.5 11.051H6.926l.723-2.72 1.166-.825-.874 3.264L6.003 18h1.569l1.83-5.661 1.526 1.269.314 4.392h1.528v-5.02l-1.757-1.756.473-1.767a3.753 3.753 0 0 0 2.853 1.313h.314V9.515a2.51 2.51 0 0 1-2.51-2.51z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </li>
  <li
    className="jsx-1448366035 kirk-itinerary-location kirk-itinerary--arrival"
    itemProp="location"
    itemScope={true}
    itemType="http://schema.org/Place"
  >
    <meta
      className="jsx-1448366035"
      content="Bordeaux"
      itemProp="name"
    />
    <meta
      className="jsx-1448366035"
      content="Bordeaux"
      itemProp="address"
    />
    <div
      className="jsx-1448366035 kirk-itinerary-location-wrapper"
    >
      <time
        className="jsx-1448366035"
        dateTime="2017-12-11T15:00"
      >
        <span
          className="kirk-text kirk-text-titleStrong c0"
        >
          15:00
        </span>
      </time>
      <div
        className="jsx-1448366035 kirk-itinerary-location-city"
      >
        <span
          className="kirk-text kirk-text-titleStrong c0"
        >
          Bordeaux
        </span>
        <div
          className="jsx-1448366035"
        >
          <div
            className="kirk-proximity c1"
          >
            <svg
              aria-hidden={true}
              className="kirk-icon c2"
              fill="#EDEDED"
              height="20px"
              viewBox="0 0 20 20"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <path
                d="M10 20C4.477 20 0 15.523 0 10S4.477 0 10 0s10 4.477 10 10-4.477 10-10 10zm1.202-14.863a1.568 1.568 0 1 0 0-3.136 1.568 1.568 0 0 0 0 3.136zm.94 1.868h-.005a1.568 1.568 0 0 0-1.563-1.554h-.007c-2.252 0-3.896 1.32-4.288 2.692L5.5 11.051H6.926l.723-2.72 1.166-.825-.874 3.264L6.003 18h1.569l1.83-5.661 1.526 1.269.314 4.392h1.528v-5.02l-1.757-1.756.473-1.767a3.753 3.753 0 0 0 2.853 1.313h.314V9.515a2.51 2.51 0 0 1-2.51-2.51z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="kirk-icon c2"
              fill="#EDEDED"
              height="20px"
              viewBox="0 0 20 20"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <path
                d="M10 20C4.477 20 0 15.523 0 10S4.477 0 10 0s10 4.477 10 10-4.477 10-10 10zm1.202-14.863a1.568 1.568 0 1 0 0-3.136 1.568 1.568 0 0 0 0 3.136zm.94 1.868h-.005a1.568 1.568 0 0 0-1.563-1.554h-.007c-2.252 0-3.896 1.32-4.288 2.692L5.5 11.051H6.926l.723-2.72 1.166-.825-.874 3.264L6.003 18h1.569l1.83-5.661 1.526 1.269.314 4.392h1.528v-5.02l-1.757-1.756.473-1.767a3.753 3.753 0 0 0 2.853 1.313h.314V9.515a2.51 2.51 0 0 1-2.51-2.51z"
              />
            </svg>
            <svg
              aria-hidden={false}
              className="kirk-icon c2"
              fill="#F78B00"
              height="20px"
              viewBox="0 0 20 20"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Pick up point is quite far fom your place
              </title>
              <path
                d="M10 20C4.477 20 0 15.523 0 10S4.477 0 10 0s10 4.477 10 10-4.477 10-10 10zm1.202-14.863a1.568 1.568 0 1 0 0-3.136 1.568 1.568 0 0 0 0 3.136zm.94 1.868h-.005a1.568 1.568 0 0 0-1.563-1.554h-.007c-2.252 0-3.896 1.32-4.288 2.692L5.5 11.051H6.926l.723-2.72 1.166-.825-.874 3.264L6.003 18h1.569l1.83-5.661 1.526 1.269.314 4.392h1.528v-5.02l-1.757-1.756.473-1.767a3.753 3.753 0 0 0 2.853 1.313h.314V9.515a2.51 2.51 0 0 1-2.51-2.51z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </li>
</ul>
`;
